src/simcore_service_deployment_agent/git_url_watcher.py

```diff
@@ -115,6 +115,29 @@ async def _git_fetch(directory: str):
     await run_cmd_line(cmd, f"{directory}")
 
 
+async def _git_get_latest_matching_tag_capture_groups(
+    directory: str, regexp: str
+) -> Optional[tuple[str]]:  # pylint: disable=unsubscriptable-object
+    cmd = [
+        "git",
+        "tag",
+        "--list",
+        "--sort=creatordate",  # Sorted ascending by date
+    ]
+    all_tags = await run_cmd_line(cmd, f"{directory}")
+    if all_tags == None:
+        return None
+    all_tags = all_tags.split("\n")
+    all_tags = [tag for tag in all_tags if tag != ""]
+    list_tags = [tag for tag in all_tags if re.search(regexp, tag) != None]
```

Comment:
TIP: what about using something like ``list_tags = re.findall( regexp, "  ".join(all_tags) ) `` SEE https://docs.python.org/3/library/re.html#finding-all-adverbs

Reply:
added this, thx